puz.py

```diff
@@ -37,12 +37,13 @@
 MASKSTRING = 'ICHEATED'
 
 ENCODING = 'ISO-8859-1'
+ENCODING_UTF8 = 'UTF-8'
+ENCODING_ERRORS = 'strict'  # raises an exception for bad chars; change to 'replace' for laxer handling
 
-ACROSSDOWN = 'ACROSS&DOWN'
+ACROSSDOWN = b'ACROSS&DOWN'
 
 BLACKSQUARE = '.'
-
-BLOCKS = '.:'
```

Comment:
I think the key changes Saul made to puz.py and the source puzpy are summarised here: https://github.com/century-arcade/xd/commit/5f0b8199c036094f958fa9863d1e4bd8fd8e8b38#diff-4be683d8935476135b8078b54bb07abd624626ee64a0838fe720a232b99b6b32

The variable `BLACKSQUARE2` did not exist in puzpy 0.2.1. It was added in the [last year](https://github.com/alexdej/puzpy/commit/0b04589346cf0b54556bf267c03479b258ebc929). I believe with its addition it handles the case that his custom variable `BLOCKS` was covering. It could be that we can return to installing `puzpy` from PyPI, if we version pin it to v0.2.3. 

@saulpw does this ring any bells?

Reply:
Alas, no memory here.  It must have been for some exotic .puz file or something.  The `assert` though would have obviated that so I'm not sure what the point was.